src/spartan/polys/multilinear.rs

```diff
@@ -325,4 +341,36 @@ mod tests {
     test_evaluation_with::<provider::bn256_grumpkin::bn256::Scalar>();
     test_evaluation_with::<provider::secp_secq::secp256k1::Scalar>();
   }
+
+  fn to_bits_be<F: PrimeField>(bit_count: usize, mut n: usize) -> Vec<F> {
+    let mut bits = Vec::with_capacity(bit_count);
+
+    for _ in 0..bit_count {
+      bits.push(if (n & 1) == 1 { F::ONE } else { F::ZERO });
+      n /= 2;
+    }
+    bits.reverse();
+    bits
+  }
+
+  fn test_layout_with<F: PrimeField>(s: usize) {
+    let rng = &mut rand::rngs::OsRng;
+    let n = 1 << s;
+    let evals = (0..n).map(|_| F::random(rng.clone())).collect::<Vec<_>>();
+
+    let p = MultilinearPolynomial::new(evals.clone());
+
+    for (i, eval) in evals.iter().enumerate() {
+      let bits = to_bits_be(s, i);
+
+      let e = p.evaluate(&bits);
+
+      assert_eq!(&e, eval);
+    }
+  }
+
+  #[test]
+  fn test_layout() {
```

Comment:
https://github.com/lurk-lab/arecibo/pull/128 was meant to make this explicit as well.